Add test_cpp_extensions tests for stream_and_event and mita_backend

egienvalue opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123614
* #123612
* #123611

Test the generic torch.Stream/Event with fake device gurad and hooks. Since we added a fake device backend, it is mutual exclusive to other backends. Tests will be skipped if TEST_CUDA or TEST_ROCM is true.

egienvalue committed:
Add test_cpp_extensions_stream_and_event

Test the generic torch.Stream/Event with fake device gurad and hooks.
ghstack-source-id: 221771109
@exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

albanD reviewed (approved):
Super cool!

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

malfet commented on test/test_cpp_extensions_stream_and_event.py:
```diff
@@ -0,0 +1,108 @@
+# Owner(s): ["module: unknown"]
```
We should not add new unowned tests to CI

albanD replied:
I created a "module: mtia" (https://github.com/pytorch/pytorch/labels/module%3A%20mtia) that you should use there.
Let's discuss offline how to make sure someone is looking at it!

malfet commented on test/test_cpp_extensions_stream_and_event.py:
```diff
@@ -0,0 +1,108 @@
+)
+@torch.testing._internal.common_utils.markDynamoStrictTest
+class TestCppExtensionStreamAndEvent(common.TestCase):
+    """Tests Stream and Event with C++ extensions."""
+
+    module = None
+
+    def setUp(self):
+        super().setUp()
+        # cpp extensions use relative paths. Those paths are relative to
+        # this file, so we'll change the working directory temporarily
+        self.old_working_dir = os.getcwd()
+        os.chdir(os.path.dirname(os.path.abspath(__file__)))
+
+    def tearDown(self):
+        super().tearDown()
+        # return the working directory (see setUp)
+        os.chdir(self.old_working_dir)
+
+    @classmethod
+    def tearDownClass(cls):
+        remove_build_path()
+
+    @classmethod
+    def setUpClass(cls):
+        remove_build_path()
+        build_dir = tempfile.mkdtemp()
+        # Load the fake device guard impl.
+        src = f"{os.path.abspath(os.path.dirname(__file__))}/cpp_extensions/mtia_extension.cpp"
+        cls.module = torch.utils.cpp_extension.load(
+            name="mtia_extension",
+            sources=[src],
+            build_directory=build_dir,
+            extra_include_paths=[
+                "cpp_extensions",
+                "path / with spaces in it",
+                "path with quote'",
+            ],
+            is_python_module=False,
+            verbose=True,
```
It feels like this code is verbatim copy from another file, is this really needed (I.e. can both tests perhaps live in the same file?)

egienvalue replied:
The test_cpp_extension_mtia_backend.py one ? They have different purposes, but they are using same cpp extension. I can create the shared cpp extension @malfet, how do you think ? 

test_cpp_extension_mtia_backend.py -> test torch.mtia module.
test_cpp_extension_stream_and_event.py -> test the general torch.Stream/Event APIs.

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions_stream_and_event"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions tests for stream_and_event and mita_backend"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions tests for stream_and_event and mita_backend"


Test the generic torch.Stream/Event with fake device gurad and hooks.
exported-using-ghexport

Differential Revision: [D55902506](https://our.internmc.facebook.com/intern/diff/D55902506/)

[ghstack-poisoned]

egienvalue commented:
@egienvalue has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56443358).

egienvalue committed:
Update on "Add test_cpp_extensions tests for stream_and_event and mita_backend"


Test the generic torch.Stream/Event with fake device gurad and hooks.

Differential Revision: [D56443358](https://our.internmc.facebook.com/intern/diff/D56443358)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions tests for stream_and_event and mita_backend"


Test the generic torch.Stream/Event with fake device gurad and hooks.

Differential Revision: [D56443358](https://our.internmc.facebook.com/intern/diff/D56443358)

[ghstack-poisoned]

egienvalue committed:
Update on "Add test_cpp_extensions tests for stream_and_event and mita_backend"


Test the generic torch.Stream/Event with fake device gurad and hooks.

Differential Revision: [D56443358](https://our.internmc.facebook.com/intern/diff/D56443358)

[ghstack-poisoned]

egienvalue commented:
@pytorchbot merge

egienvalue committed:
Update on "Add test_cpp_extensions tests for stream_and_event and mita_backend"


Test the generic torch.Stream/Event with fake device gurad and hooks.



[ghstack-poisoned]

egienvalue commented:
I am relanding the PR. All ROCm signals look good. 
cc @jeffdaily  @jithunnair-amd
